flocker/node/agents/ebs.py

```diff
@@ -0,0 +1,198 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+An EBS implementation of the ``IBlockDeviceAPI``.
+"""
+
+from bitmath import Byte, GB
+import time
+from uuid import UUID
+
+from boto import ec2
+from pyrsistent import PRecord, field
+from zope.interface import implementer
+
+from .blockdevice import IBlockDeviceAPI, BlockDeviceVolume, UnknownVolume
+
+DATASET_ID_LABEL = u'flocker-dataset-id'
+METADATA_VERSION_LABEL = u'flocker-metadata-version'
+CLUSTER_ID_LABEL = u'flocker-cluster-id'
+
+
+def ec2_client(region, zone, access_key_id, secret_access_key):
+    """
+    Establish connection to EC2 client.
+
+    :param String region: The name of the EC2 region to connect to.
+    :param String zone: The zone for the EC2 region to connect to.
+    :param String access_key_id: "aws_access_key_id" credential for EC2.
+    :param String secret_access_key: "aws_secret_access_key" EC2 credential.
+
+    :return: An ``_EC2`` giving information about EC2 client connection
+        and EC2 instance zone.
+    """
+    return _EC2(zone=zone,
+                connection=ec2.connect_to_region(
+                    region,
+                    aws_access_key_id=access_key_id,
+                    aws_secret_access_key=secret_access_key))
+
+
+class _EC2(PRecord):
+    """
+    :ivar string zone: The name of the zone for the connection.
+    :ivar EC2Connection connection: Object representing connection to
+        an EC2 instance.
+    """
+    zone = field()
+    connection = field()
+
+
+@implementer(IBlockDeviceAPI)
+class EBSBlockDeviceAPI(object):
+    """
+    An EBS implementation of ``IBlockDeviceAPI`` which creates
+    block devices in an EC2 cluster using Boto APIs.
+    """
+    def __init__(self, ec2_client, cluster_id):
+        """
+        Initialize EBS block device API instance.
+
+        :param ``_EC2`` ec2_client: PRecord storing EC2
+            client connection and EC2 instance zone.
```

Comment:
Probably just need to specify the type and say something like "A record of EC2 connection and zone."